src/card-list.js

```diff
@@ -1,28 +1,47 @@
 import React from 'react'
-import Card from './card.js'
+import NoCards from './no-cards.js'
+import Cards from './cards'
+import FilterDrop from './filter-dropdown.js'
 
-export default function CardList(props) {
-  return (
-    <React.Fragment>
-      {props.cards.length === 0 && <div className="text-center mt-5">
-        <h2>You have no cards saved</h2>
-        <a className="btn btn-primary mt-3" role="button" href="#new">
-        Make a Card
-        </a>
-      </div>}
-      {props.cards.length > 0 &&
-      <div className="container-fluid">
-        <div className="row justify-content-start">
-          {props.cards.map((card) => {
-            return <Card
-              id={card.cardId}
-              key={card.cardId}
-              remove={props.remove}
-              card={card}/>
-          })}
-        </div>
-      </div>
-      }
-    </React.Fragment>
-  )
+export default class CardList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {show: 'none'}
+    this.sort = this.sort.bind(this)
+  }
+  sort(event) {
+    if (!event) {
+      this.setState({show: 'none'})
+    }
+    else {
+      this.setState({show: event.target.value})
+    }
+  }
+  render() {
+    const topics = []
+    this.props.cards
+      .filter((card) => {
+        if (!topics.includes(card.topic)) {
+          topics.push(card)
+        }
+      })
+    const renderCards = this.props.cards
+      .filter((card) => {
+        if (this.state.show !== 'none') {
+          return card.topic === this.state.show
+        }
+        return card
+      })
+    return (
+      <React.Fragment>
+        {this.props.cards.length === 0 && <NoCards/>}
+        {this.props.cards.length > 0 &&
+          <React.Fragment>
+            <FilterDrop topics={topics} sort={this.sort}/>
```

Comment:
Topics is actually an array of flash card objects. Can you compute an array of just the topics?